### ISSUE TITLE
Need to implement gradient descent to optimize pre-defined weights 

### ISSUE THREAD

**Issue by oysters76** [oysters76/neural-net-backprop#3]
Currently, the weights and the inputs are pre-defined under the 'values' object, need a way to optimize the weights to minimize a trivial loss function (y-y_hat)^2
